### ISSUE TITLE
How to suppress parser/current warnings?

### ISSUE THREAD

**Issue by firedev** [bbatsov/rubocop#1819]
When I run rubocop I get something like this on top:
```
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.6-compliant syntax, but you are running 2.1.2.
```
Is there a way to suppress that? It breaks Syntastic for Vim.

I can't change ruby version. There are other people working on this project and the version is locked.

**Comment by bbatsov** [comment]
You can't. It comes from the `parser` gem. /cc @whitequark

So many people have complaint about this message that I'm starting to think it's more trouble that it's worth.

**Comment by firedev** [comment]
I think it would made sense if it was displayed with `--verbose` or `--debug`

**Comment by bbatsov** [comment]
As I said - it's not in RuboCop - we can suppress the output, but I don't want to do this, as we might end up suppressing something useful.

**Comment by whitequark** [comment]
2.1.2 and 2.1.6 genuinely differ in syntax.

If you want to complain to someone, compare to the Ruby MRI authors, who consider these changes between patchlevels acceptable.

I will not change `parser`.

**Comment by firedev** [comment]
Err... Okay. So how do I force Rubocop to not complaint about that?

**Comment by whitequark** [comment]
Run Rubocop as, for example, `ruby -W0 -S rubocop args...`.

**Comment by bbatsov** [comment]
Keep in mind this will kill all the warnings, not just this one.

**Comment by firedev** [comment]
Thanks @whitequark I just deleted the warnings.

**Comment by ioquatix** [comment]
@whitequark if I can get some kind of unofficial guarantee that the Ruby core team won't break syntax in patch releases going forward, can we remove the warning?

**Comment by whitequark** [comment]
@ioquatix I'm not sure why you mean by "unofficial". If the Ruby core team publicly commits to not changing syntax in patch releases (which will doubtlessly benefit the entire ecosystem) I am of course fine with removing the warning, since there would be no reason to have it anymore.

**Comment by ioquatix** [comment]
@whitequark I don't know what's possible yet, and I don't know at what level we could make the guarantee, ideally it would never happen, but I need to investigate further, e.g. why it happened in the past. I guess I'm just looking at options to try and help mitigate the need for the warning.

**Comment by whitequark** [comment]
@ioquatix I think that is an excellent initiative and I look forward to your results.

**Comment by westonganger** [comment]
Just use the `warning` gem to suppress the output. Its what we must do I suppose.


```ruby
# config/initializers/warning_ignores.rb

### https://github.com/jeremyevans/ruby-warning

Warning.ignore(/warning: parser/)
```

**Comment by westonganger** [comment]
For anyone else coming here looking for a solution for now. Just use the `warning` gem to suppress the output. Its what we must do I suppose.


```ruby
# config/initializers/warning_ignores.rb

### https://github.com/jeremyevans/ruby-warning

Warning.ignore(/warning: parser/)
```

**Comment by marcandre** [comment]
If I'm not mistaken, you should simply always use the latest `parser` gem and the latest Ruby. They are normally in sync.

**Comment by vizcay** [comment]
> For anyone else coming here looking for an easy solution without updating their Ruby version. Just use the `warning` gem to suppress the output.
> 
> ```ruby
> # config/initializers/warning_ignores.rb
> 
> ### https://github.com/jeremyevans/ruby-warning
> 
> Warning.ignore(/warning: parser/)
> ```

@westonganger  This allows to silence the warning starting rails server, etc. But not when just running rubocop.. or I'm missing something?
